Provide configuration to SpatieTagsInput to allow original behaviour of suggestions for tags with types

sten opened the pull request:
@danharrin as promised in https://github.com/filamentphp/filament/pull/7374 I added an extra configuration flag to the SpatieTagsInput to be able to overwrite the old default behaviour with the new implementation.

Do you think the naming of the config variable `disableTypeCheckingForSuggestions` is clear?

- [x] Changes have been thoroughly tested to not break existing functionality.
- [ ] New functionality has been documented or existing documentation has been updated to reflect changes.
- [ ] Visual changes are explained in the PR description using a screenshot/recording of before and after.

sten committed:
Provide configuration to SpatieTagsInput to allow original behaviour of suggestions for tags with types

danharrin commented:
How about this API

`SpatieTagsInput::make()` - suggestions from all types
`SpatieTagsInput::make()->type('blog')` - suggestions only from `blog` type
`SpatieTagsInput::make()->type(null)` - suggestions only from `null` type

sten committed:
Improve configuration for tag types

Implemented suggested API

sten commented:
@danharrin I implemented your suggested API. Because `$type` is set to `null` on object creation, I added another var to be able to check if `$type` was forcefully set to `null`.

danharrin commented:
Maybe we could set the default of that prop to be `new AllTypes()` where AllTypes is some empty object that represents that state

sten committed:
Add AllTagTypes type and make column, input and info list work

- Added AllTagTypes class to model that any type is allowed
- Implementation for the input component, infolist component and the table column.
- I added a new tag sync function, syncTagsWithAnyType in SpatieTagsInput, because the tag sync functions of the spatie package create new tags if the type is not available for look up.

@danharrin Let me know your feedback, please.

sten committed:
fix phpstan errors on SpatieTagsInput

sten committed:
fix phpstan errors

fixing typing errors for phpstan check

sten commented:
@danharrin I added a new type as you suggested to be able to set the input, infolist and column to allow any type. 

@danharrin Thank you for the feedback! I had to add a function to sync the tags to the model to avoid creating duplicate tags without a type if the type already existed, because the HasTags trait did not have this functionality.

Let me know what you think.

danharrin committed:
wip

danharrin committed:
Update SpatieTagsInput.php

danharrin commented:
Thanks @sten, this is exactly what we need. You interpreted my requests really well
